pkgs/development/libraries/mlpack/default.nix

```diff
@@ -0,0 +1,28 @@
+{ stdenv, lib, armadillo, boost, cmake, fetchFromGitHub, ensmallen
+, enableOpenMP ? false }:
+
+stdenv.mkDerivation rec {
+  name = "mlpack-${version}";
+  version = "3.2.1";
+
+  src = fetchFromGitHub {
+    owner = "mlpack";
+    repo = "mlpack";
+    rev = version;
+    sha256 = "sha256-/UN/Pmt7m3s9pq4I3XTt0AbHEj3nbbdWeLLcs1TE+9w=";
+  };
+
+  buildInputs = [ cmake armadillo boost ensmallen ];
+
+  cmakeFlags = [ "-DUSE_OPENMP=${if enableOpenMP then "ON" else "OFF"}" ];
```

Comment:
Just making sure:
Here you don't need to include openmp in the buildINputs because it's provided through ensmallen?

Reply:
Right

Reply:
Then we should also put an assert in this file, to make sure that ensmallen is passed (`!=null`) whenever `enableOpenMP` is true.

Reply:
hm well ensmallen is needed anyway, if enableOpenMP is true then we need to have ensmallenOpenMP as dependency instead of the regular one.